pkg/extension/sumologicextension/extension.go

```diff
@@ -40,9 +46,38 @@ type SumologicExtension struct {
 	closeOnce        sync.Once
 }
 
+type FullRegisterKeyResponse struct {
+	CollectorID   int64         `json:"collectorId"`
+	CredentialID  string        `json:"credentialId"`
+	CredentialKey string        `json:"credentialKey"`
+	CustomerID    int64         `json:"customerId"`
+	Errors        []interface{} `json:"errors"`
+	HTTPCode      int64         `json:"httpCode"`
+	Warnings      []interface{} `json:"warnings"`
+}
```

Comment:
I was looking at linters in golangci-lint and it seems that `deadcode` doesn't detect exported types as unused :/ and [it's not configurable](https://github.com/remyoudompheng/go-misc/blob/2d6ac652a50e368c874b15d00ea5e346e240e906/deadcode/deadcode.go#L114)

Can you remove this and check other structs as well?